test/unit/org/apache/cassandra/config/DatabaseDescriptorTest.java

```diff
@@ -590,4 +591,32 @@ public void testApplyTokensConfigInitialTokensOneNumTokensNotSet()
         Assert.assertEquals(Integer.valueOf(1), config.num_tokens);
         Assert.assertEquals(1, DatabaseDescriptor.tokensFromString(config.initial_token).size());
     }
+
+    @Test
+    public void testDenylistInvalidValuesRejected()
+    {
+        DatabaseDescriptor.loadConfig();
+
+        expectIllegalArgumentException(DatabaseDescriptor::setDenylistRefreshSeconds, 0);
+        expectIllegalArgumentException(DatabaseDescriptor::setDenylistRefreshSeconds, -1);
+
+        expectIllegalArgumentException(DatabaseDescriptor::setDenylistKeysPerTableMax, 0);
+        expectIllegalArgumentException(DatabaseDescriptor::setDenylistKeysPerTableMax, -1);
+
+        expectIllegalArgumentException(DatabaseDescriptor::setDenylistKeysTotalMax, 0);
+        expectIllegalArgumentException(DatabaseDescriptor::setDenylistKeysTotalMax, -1);
+    }
+
+    private void expectIllegalArgumentException(Consumer<Integer> c, int val)
```

Comment:
I feel we need to use `assertThatThrownBy`, easily lets to check not only that an exception was thrown, but also its message. It could help you to catch this issue for example: https://github.com/apache/cassandra/pull/1213#discussion_r722979906